### PR TITLE
Feature/417 leftout comments

### PR DESCRIPTION
**Pull request by baristaGeek**
## Description
Adds detecting multi-line leftover comments as a code smell

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC
- [ ] Test
## Notes
I tested with the commit that this file has that has a leftover comment itself (already removed) and it successfully dects the leftover comment. 

However, there is an error in the logic. The code gets the correct position of the occurrence of the leftover comment (number of lines below the @@ header hunk, not the actual line number in the file). However it comments the lineNumber under the wrong header hunk (the first one). Thsi needs to be fixed for this to be upgraded from draft PR. 

## Acceptance
- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by EstebanDalelR — COMMENTED** [review]
What can we extract? This is a 600 LOC file!
Code seems ok, but I think we do not want blocking commenting of files but rather choosing one?

**Comment by baristaGeek** [comment]
> What can we extract? This is a 600 LOC file! Code seems ok, but I think we do not want blocking commenting of files but rather choosing one?

I totally agree there's a bunch of things to abstract. I'm worrying about making it work first, then I'll worry about abstracting the code.

**Comment by baristaGeek** [comment]
**I upgraded the PR to "ready for review". Changes include:**
- We now handle the multi-line comments created with "//" on each line case. In addition to the actual multi-line syntax (/* */) case. Image below. 
- We need to index lineNumbers at 4 instead of 0 because the GitHub UI always adds 3 lines below the @@ header (plus they are indexed at 1 instead of 0, plus the octokit comments.position parameter is the number of lines below the @@ heaer instead of the actual line number in the file). 
- I changed the name of the file and the function to detectCodeSmells since we're generalizing the capabilities of this feature. 

**Considerations**
- Might be missing syntax from some major languages
- I agree that the file can be abstracted. I propose we agree on whether this works or not, and then the abstraction step could happen in a separate PR. 

CC: @EstebanDalelR 

Screenshot 2024-01-23 at 4 05 03 PM

File: utils/actions/detectCodeSmells.ts [hunk 1]
```diff
@@ -79,11 +81,61 @@ export default async function detectConsoleLogs({
...
       });
   }
+
   const latestCommitHash = await getLatestCommitHash();
 
   const commentPromises = diffFiles.map(async (file) => {
     const { additions } = getLineDiffs(file.patch ?? "");
 
+    // Leftover comment RegEx
+    // const leftoverCommentRegex = /^\/\*[\s\S]*?\*\//gm;
```
Comment (EstebanDalelR) [inline utils/actions/detectCodeSmells.ts:91]: This seems very independent, can we split it as to call it in this function?
Other than that LGTM!

Comment (baristaGeek) [reply utils/actions/detectCodeSmells.ts:91]: Lol we needed the code smell itself. Will remove this!

**Review by EstebanDalelR — APPROVED** [review]

**Pull request merged**